docs/Makefile

```diff
@@ -1,77 +1,89 @@
+BUILDDIR      = _build
+VENVDIR       = $(SPHINXDIR)/venv
+VENV          = $(VENVDIR)/bin/activate
 
+.PHONY: help woke-install install run html epub serve clean clean-doc \
+        spelling linkcheck woke Makefile
 
 # Put it first so that "make" without argument is like "make help".
-help:
-	@echo "The following make targets are available:"
-	@echo "* install   -- set up a virtual-env to work on documentation"
-	@echo "* run       -- watch, build and serve the documentation"
-	@echo "* html      -- build HTML output under $(BUILDDIR)/html"
-	@echo "* serve     -- serve HTML output from $(BUILDDIR)/html"
-	@echo "* epub      -- build epub output under $(BUILDDIR)/epub"
-	@echo "* pdf       -- build pdf output under $(BUILDDIR)/pdf"
-	@echo "* clean-doc -- clean built doc files"
-	@echo "* clean     -- clean full environment"
-	@echo "* spelling  -- check spelling"
-	@echo "* linkcheck -- check external links"
-	@echo "* woke      -- check inclusive language"
-	@echo "-------------------------------------------"
-
-install:
-	@echo "... setting up virtualenv"
-	$(PYTHON) -m venv $(VENVDIR)
-	$(IN_VENV); pip install --upgrade -r .sphinx/requirements.txt
-
-run:
-	$(IN_VENV); sphinx-autobuild -b html "$(SOURCEDIR)" "$(BUILDDIR)"/html
-
-html:
-	$(IN_VENV); $(SPHINXBUILD) -b html "$(SOURCEDIR)" "$(BUILDDIR)"/html
-
-epub:
-	$(IN_VENV); $(SPHINXBUILD) -b epub "$(SOURCEDIR)" "$(BUILDDIR)"/epub
+help: $(VENVDIR)
+	@. $(VENV); $(SPHINXBUILD) -M help "$(SOURCEDIR)" "$(BUILDDIR)" $(SPHINXOPTS) $(O)
 
-pdf:
-	$(IN_VENV); $(SPHINXBUILD) -b latex "$(SOURCEDIR)" "$(BUILDDIR)"/pdf
-	$(IN_VENV); $(MAKE) -C "$(BUILDDIR)"/pdf all-pdf
+# Explicit target avoids fall-through to the "Makefile" target.
+$(SPHINXDIR)/requirements.txt:
+	test -f $(SPHINXDIR)/requirements.txt
 
-serve:
-	$(PYTHON) -m http.server 8000 -d "$(BUILDDIR)"/html
+# If requirements are updated, venv should be rebuilt and timestamped.
+$(VENVDIR): $(SPHINXDIR)/requirements.txt
+	@echo "... setting up virtualenv"
+	python3 -m venv $(VENVDIR)
+	. $(VENV); pip install --require-virtualenv \
+	    --upgrade -r $(SPHINXDIR)/requirements.txt \
+            --log $(VENVDIR)/pip_install.log
+	@test ! -f $(VENVDIR)/pip_list.txt || \
+            mv $(VENVDIR)/pip_list.txt $(VENVDIR)/pip_list.txt.bak
+	@. $(VENV); pip list --local --format=freeze > $(VENVDIR)/pip_list.txt
+	@echo "\n" \
+        "--------------------------------------------------------------- \n" \
+        "* watch, build and serve the documentation: make run \n" \
+        "* only build: make html \n" \
+        "* only serve: make serve \n" \
+        "* clean built doc files: make clean-doc \n" \
+        "* clean full environment: make clean \n" \
+        "* check links: make linkcheck \n" \
+        "* check spelling: make spelling \n" \
+        "* check inclusive language: make woke \n" \
+        "* other possible targets: make <press TAB twice> \n" \
+        "--------------------------------------------------------------- \n"
```

Comment:
Question: Why is this here?

I am aware that `$(VENVDIR)` is called by help, but when the folder already exists (aka after the first help call), this will not be displayed again. This looks like bad UX to me.

Reply:
And badly formatted, too ;)

Anyway, I'm aware that the Starter Pack isn't perfect yet, but the idea is to unify on its use, so that future updates and docs maintenance is simple across the board. As we update the SP, changes will be propagated to projects that have adopted it.

IOW, would you be OK with raising this as an issue for the SP?

Reply:
That's fine for me. :+1: